client/src/js/components/bhReceipt.js

```diff
@@ -0,0 +1,34 @@
+var bhReceiptTemplate =
+  '<a ng-click="$ctrl.open()" href="">{{ $ctrl.displayValue }}</a>';
+
+angular.module('bhima.components')
+  .component('bhReceipt', {
+    template : bhReceiptTemplate,
+    controller  : bhReceiptController,
+    bindings    : {
+      value : '<',
+      displayValue : '<',
+      type : '@',
+    },
+  });
+
+bhReceiptController.$inject = ['ReceiptModal', '$log'];
+
+function bhReceiptController(ReceiptModal, $log) {
+  var $ctrl = this;
+
+  $ctrl.$onInit = function $onInit() {
+    // make sure the receipt type exists before it is clicked
+    var hasCallbackFn = ReceiptModal[$ctrl.type];
+    if (!hasCallbackFn) {
+      $log.warn('Warning: Cannot find '
```

Comment:
Ha ha this is actually an error. This is a no blocking comment.

Reply:
Yeah, you are right.  I'll make it an error.